docs/contributing/target-framework-strategy.md

```diff
@@ -0,0 +1,122 @@
+# Target Framework Strategy
+
+## Layers
+
+The roslyn repository produces components for a number of different products that push varying ship and TFM constraints on us. A summary of some of our dependencies are : 
+
+- Build Tools: requires us to ship compilers on `net472`
+- .NET SDK: requires us to ship compilers on current servicing target framework (presently `net8.0`)
+- Source build: requires us to ship `$(NetCurrent)` and `$(NetPrevious)` in workspaces and below (presently `net9.0` and `net8.0` respectively)
```

Comment:
Seems this is no longer true?

Reply:
Split this out into both repo and full source build discussion